pkg/fn/eval_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fn
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+)
+
+func TestNotFound(t *testing.T) {
+	var err error
+
+	fn := &v1.Function{Image: "foo"}
+	err = &NotFoundError{Function: *fn}
+
+	var notFoundErr *NotFoundError
+	if !errors.As(err, &notFoundErr) {
+		t.Fatalf("expected NotFoundError to satisfy errors.As")
+	}
+
+	if got, want := notFoundErr.Function, *fn; !reflect.DeepEqual(got, want) {
```

Comment:
for such comparisons, I'd recommend using https://github.com/google/go-cmp
It comes with good Diff algorithm too which can be used to generate more helpful error messages.

Example: https://github.com/GoogleContainerTools/kpt/blob/main/porch/engine/pkg/kpt/render_test.go#L76